mne/forward/tests/test_make_forward.py

```diff
@@ -186,41 +225,76 @@ def test_make_forward_solution_bti(fname_src_small):
     _compare_forwards(fwd, fwd_py, 248, n_src_small)
 
 
-@requires_mne
-def test_make_forward_solution_ctf(tmp_path, fname_src_small):
-    """Test CTF w/compensation against C."""
+@pytest.mark.parametrize('other', [
+    pytest.param('MNE-C', marks=requires_mne_mark()),
+    pytest.param('openmeeg', marks=requires_openmeeg_mark()),
+])
+def test_make_forward_solution_ctf(tmp_path, fname_src_small, other):
+    """Test CTF w/compensation against MNE-C or OpenMEEG."""
     src = read_source_spaces(fname_src_small)
-    fwd_py = make_forward_solution(fname_ctf_raw, fname_trans, src,
-                                   fname_bem_meg, eeg=False, verbose=True)
-
-    fwd = _do_forward_solution('sample', fname_ctf_raw, mri=fname_trans,
-                               src=fname_src_small, bem=fname_bem_meg,
-                               eeg=False, meg=True, subjects_dir=subjects_dir)
-    _compare_forwards(fwd, fwd_py, 274, n_src_small)
+    raw = read_raw_fif(fname_ctf_raw)
+    assert raw.compensation_grade == 3
+    if other == 'openmeeg':
+        mindist = 20.
+        n_src_want = 51
+    else:
+        assert other == 'MNE-C'
+        mindist = 0.
+        n_src_want = n_src_small == 108
+    mindist = 20. if other == 'openmeeg' else 0.
+    fwd_py = make_forward_solution(
+        fname_ctf_raw, fname_trans, src, fname_bem_meg, eeg=False,
+        mindist=mindist, verbose=True)
+
+    if other == 'openmeeg':
+        # TODO: This should be a 1-layer, but it's broken
+        # (some correlations become negative!)...
+        bem_surfaces = read_bem_surfaces(fname_bem)  # fname_bem_meg
+        bem = make_bem_solution(bem_surfaces, solver='openmeeg')
+        # TODO: These tolerances are bad
+        tol_kwargs = dict(meg_atol=1, meg_corr_tol=0.65, meg_rdm_tol=0.6)
```

Comment:
@agramfort I added a test for CTF data. The results there are less good, but maybe okay? (I had to use a 3-layer as the 1-layer is also terrible here as well, some correlations in that case are negative...)

Reply:
it's possible the bug in Python wrapper with ref / shared pointers affects our results here. I fear we need to fix it there first. Some alternative to move faster here is to work by writing files on disk and reading them without numpy arrays

Reply:
Using files for the geom at least does not help with these tols